File: japiwdbe/WdbeQCvrList.java
/**
  * \file WdbeQCvrList.h
  * Java API code for record of table TblWdbeQCvrList
  * \author Alexander Wirthmueller
  * \date created: 11 Jul 2020
  * \date modified: 11 Jul 2020
  */

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQCvrList {
	
	public WdbeQCvrList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String stubRefWdbeMCoreproject
				, int Major
				, int Minor
				, int Sub
				, String stubBcvRefWdbeMCoreversion
				, String srefIxVState
				, String titIxVState
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.stubRefWdbeMCoreproject = stubRefWdbeMCoreproject;
		this.Major = Major;
		this.Minor = Minor;
		this.Sub = Sub;
		this.stubBcvRefWdbeMCoreversion = stubBcvRefWdbeMCoreversion;
		this.srefIxVState = srefIxVState;
		this.titIxVState = titIxVState;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String stubRefWdbeMCoreproject;
	public int Major;
	public int Minor;
	public int Sub;
	public String stubBcvRefWdbeMCoreversion;
	public String srefIxVState;
	public String titIxVState;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQCvrList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			stubRefWdbeMCoreproject = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMCoreproject", "cpr", null, 0);
			Major = Xmlio.extractIntegerUclc(doc, basexpath, "Major", "maj", null, 0);
			Minor = Xmlio.extractIntegerUclc(doc, basexpath, "Minor", "min", null, 0);
			Sub = Xmlio.extractIntegerUclc(doc, basexpath, "Sub", "sub", null, 0);
			stubBcvRefWdbeMCoreversion = Xmlio.extractStringUclc(doc, basexpath, "stubBcvRefWdbeMCoreversion", "bcv", null, 0);
			srefIxVState = Xmlio.extractStringUclc(doc, basexpath, "srefIxVState", "ste", null, 0);
			titIxVState = Xmlio.extractStringUclc(doc, basexpath, "titIxVState", "ste2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQCvrList.cpp
/**
	* \file WdbeQCvrList.cpp
	* API code for table TblWdbeQCvrList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 23 Aug 2020
	* \date modified: 23 Aug 2020
	*/

#include "WdbeQCvrList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQCvrList
 ******************************************************************************/

WdbeQCvrList::WdbeQCvrList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string stubRefWdbeMCoreproject
			, const usmallint Major
			, const usmallint Minor
			, const usmallint Sub
			, const string stubBcvRefWdbeMCoreversion
			, const string srefIxVState
			, const string titIxVState
		) {
	this->jnum = jnum;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->stubRefWdbeMCoreproject = stubRefWdbeMCoreproject;
	this->Major = Major;
	this->Minor = Minor;
	this->Sub = Sub;
	this->stubBcvRefWdbeMCoreversion = stubBcvRefWdbeMCoreversion;
	this->srefIxVState = srefIxVState;
	this->titIxVState = titIxVState;
};

bool WdbeQCvrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQCvrList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMCoreproject", "cpr", stubRefWdbeMCoreproject);
		extractUsmallintUclc(docctx, basexpath, "Major", "maj", Major);
		extractUsmallintUclc(docctx, basexpath, "Minor", "min", Minor);
		extractUsmallintUclc(docctx, basexpath, "Sub", "sub", Sub);
		extractStringUclc(docctx, basexpath, "stubBcvRefWdbeMCoreversion", "bcv", stubBcvRefWdbeMCoreversion);
		extractStringUclc(docctx, basexpath, "srefIxVState", "ste", srefIxVState);
		extractStringUclc(docctx, basexpath, "titIxVState", "ste2", titIxVState);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQCvrList
 ******************************************************************************/

ListWdbeQCvrList::ListWdbeQCvrList() {
};

ListWdbeQCvrList::ListWdbeQCvrList(
			const ListWdbeQCvrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCvrList(*(src.nodes[i]));
};

ListWdbeQCvrList& ListWdbeQCvrList::operator=(
			const ListWdbeQCvrList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCvrList(*(src.nodes[i]));

	return *this;
};

ListWdbeQCvrList::~ListWdbeQCvrList() {
	clear();
};

void ListWdbeQCvrList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQCvrList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQCvrList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQCvrList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQCvrList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQCvrList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQCvrList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

